tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,567 @@
+{
+	const struct aics_audio_desc *cmd = (void *) data;
+	char description[cmd->desc_len + 1]; /*  one byte for null termination */
+
+	LOG_DBG("AICS description");
+
+	if (cmd->desc_len > BT_AICS_MAX_INPUT_DESCRIPTION_SIZE) {
+		LOG_ERR("Too long input (%u chars required)",
+			BT_AICS_MAX_INPUT_DESCRIPTION_SIZE);
+		goto rsp;
+	}
+
+	utf8_lcpy(description, cmd->desc, sizeof(description));
+
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		if (bt_aics_description_set(included.aics[i], description) != 0) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_DESCRIPTION,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+rsp:
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_DESCRIPTION, CONTROLLER_INDEX,
+		   BTP_STATUS_FAILED);
+}
+
+void tester_handle_aics(uint8_t opcode, uint8_t index, uint8_t *data, uint16_t len)
+{
+	switch (opcode) {
+	case AICS_READ_SUPPORTED_COMMANDS:
+		aics_supported_commands(data, len);
+		break;
+	case AICS_SET_GAIN:
+		aics_set_gain(data);
+		break;
+	case AICS_MUTE:
+		aics_mute();
+		break;
+	case AICS_UNMUTE:
+		aics_unmute();
+		break;
+	case AICS_MAN_GAIN:
+		aics_man_gain();
+		break;
+	case AICS_AUTO_GAIN:
+		aics_auto_gain();
+		break;
+	case AICS_MAN_GAIN_ONLY:
+		aics_auto_gain_only();
+		break;
+	case AICS_AUTO_GAIN_ONLY:
+		aics_auto_gain_only();
+		break;
+	case AICS_DESCRIPTION:
+		aics_desc(data);
+		break;
+	case AICS_MUTE_DISABLE:
+		aics_mute_disable();
+		break;
+	default:
+		tester_rsp(BTP_SERVICE_ID_AICS, opcode, index,
+			   BTP_STATUS_UNKNOWN_CMD);
+		break;
+	}
+}
+
+/* Volume Offset Control Service */
+static void vocs_supported_commands(void)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(BTP_DATA_MAX_SIZE);
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, VOCS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, VOCS_UPDATE_LOC);
+	net_buf_simple_add_u8(buf, VOCS_UPDATE_DESC);
+
+	tester_send(BTP_SERVICE_ID_VOCS, VOCS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, buf->data, buf->len);
+}
+
+static void vocs_state_cb(struct bt_vocs *inst, int err, int16_t offset)
+{
+	LOG_DBG("VOCS state callback err (%d)", err);
+}
+
+static void vocs_location_cb(struct bt_vocs *inst, int err, uint32_t location)
+{
+	LOG_DBG("VOCS location callback err (%d)", err);
+}
+
+static void vocs_description_cb(struct bt_vocs *inst, int err,
+				char *description)
+{
+	LOG_DBG("VOCS desctripion callback (%d)", err);
+}
+
+static struct bt_vocs_cb vocs_cb = {
+	.state = vocs_state_cb,
+	.location = vocs_location_cb,
+	.description = vocs_description_cb
+};
+
+void vocs_audio_desc(uint8_t *data)
+{
+	struct vocs_audio_desc *cmd = (void *) data;
+	char description[cmd->desc_len + 1]; /*  one byte for null termination */
+
+	LOG_DBG("VOCS description");
+
+	/* FIXME: Spec does not specify maximum desc length */
+	if (cmd->desc_len > 16) {
+		LOG_ERR("Too long input (%u chars required)", 16);
+		goto rsp;
+	}
+
+	utf8_lcpy(description, cmd->desc, sizeof(description));
```

Comment:
this function assumes src to be NULL terminated string which is not the case here, I'm not sure if there is helper for our usecase so maybe just do:
memcpy(description, cmd->desc, sizeof(description));
description[cmd->desc] = '\0';

Reply:
> this function assumes src to be NULL terminated string which is not the case here

Since `description` is defined using `cmd->desc_len + 1`, this still works as it copies up to `cmd->desc_len` octets. 

But otherwise I agree that we can just `memcpy` here, since we are not interesting in handling truncation here, and we do have a length check above.

Reply:
Done